app/src/main/java/com/github/se/polyfit/data/processor/LocalDataProcessor.kt

```diff
@@ -26,6 +28,26 @@ class LocalDataProcessor @Inject constructor(private val mealDao: MealDao) {
     return getCaloriesPerMealOnSpecificDay(LocalDate.now())
   }
 
+  fun getCaloriesPerMealOccasionTodayLiveData(): LiveData<List<Pair<MealOccasion, Double>>> {
+    return mealDao.getMealsCreatedOnOrAfterDateLiveData(LocalDate.now()).map { meals ->
+      val allOccasion =
+          MealOccasion.values().toList().filter { it != MealOccasion.OTHER }.map { Pair(it, 0.0) }
```

Comment:
shouldn't you also make sure the meal is complete ?

Reply:
Yep catch. This will avoid displaying uncompleted meal !